File: examples/sync/kucoin_public.py
import dcex


def main() -> None:
    client = dcex.kucoin()

    instruments = client.get_spot_instrument_info()
    print(instruments)

    ticker = client.get_spot_ticker(product_symbol="BTC-USDT-SPOT")
    print(ticker)


if __name__ == "__main__":
    main()

File: crates/dcex/examples/kucoin_public.rs
use std::time::Duration;

use dcex::exchanges::kucoin::KucoinClient;

#[tokio::main]
async fn main() -> dcex::Result<()> {
    let client = KucoinClient::public(Duration::from_secs(10))?;

    let instruments = client.get_spot_instrument_info().await?;
    println!("{}", instruments.data);

    let ticker = client.get_spot_ticker("BTC-USDT-SPOT").await?;
    println!("{}", ticker.data);

    Ok(())
}
